lib/web_ui/lib/src/engine/compositor/surface.dart

```diff
@@ -119,51 +121,75 @@ class Surface {
       ..position = 'absolute'
       ..width = '${logicalSize.width.ceil()}px'
       ..height = '${logicalSize.height.ceil()}px';
-    final int glContext = canvasKit.callMethod('GetWebGLContext', <dynamic>[
-      htmlCanvas,
-      // Default to no anti-aliasing. Paint commands can be explicitly
-      // anti-aliased by setting their `Paint` object's `antialias` property.
-      js.JsObject.jsify({'antialias': 0}),
-    ]);
-    _grContext =
-        canvasKit.callMethod('MakeGrContext', <dynamic>[glContext]);
-
-    if (_grContext == null) {
-      throw CanvasKitError('Could not create a graphics context.');
-    }
 
-    // Set the cache byte limit for this grContext, if not specified it will use
-    // CanvasKit's default.
-    _syncCacheBytes();
+    htmlElement = htmlCanvas;
+    if (canvasKitForceCpuOnly) {
+      return _makeSoftwareCanvasSurface(htmlCanvas);
+    } else {
+      // Try WebGL first.
+      final int glContext = canvasKit.callMethod('GetWebGLContext', <dynamic>[
+        htmlCanvas,
+        // Default to no anti-aliasing. Paint commands can be explicitly
+        // anti-aliased by setting their `Paint` object's `antialias` property.
+        js.JsObject.jsify({'antialias': 0}),
+      ]);
+
+      if (glContext == 0) {
+        return _makeSoftwareCanvasSurface(htmlCanvas);
+      }
+
+      _grContext =
+          canvasKit.callMethod('MakeGrContext', <dynamic>[glContext]);
+
+      if (_grContext == null) {
+        return _makeSoftwareCanvasSurface(htmlCanvas);
+      }
+
+      // Set the cache byte limit for this grContext, if not specified it will use
+      // CanvasKit's default.
+      _syncCacheBytes();
+
+      js.JsObject? skSurface =
+          canvasKit.callMethod('MakeOnScreenGLSurface', <dynamic>[
+        _grContext,
+        size.width,
+        size.height,
+        canvasKit['SkColorSpace']['SRGB'],
+      ]);
 
-    final js.JsObject? skSurface =
-        canvasKit.callMethod('MakeOnScreenGLSurface', <dynamic>[
-      _grContext,
-      size.width,
-      size.height,
-      canvasKit['SkColorSpace']['SRGB'],
-    ]);
+      if (skSurface == null) {
+        return _makeSoftwareCanvasSurface(htmlCanvas);
+      }
 
-    if (skSurface == null) {
-      throw CanvasKitError('Could not create a surface.');
+      return CkSurface(skSurface!, _grContext, glContext);
     }
+  }
 
-    htmlElement = htmlCanvas;
-    return CkSurface(skSurface, _grContext!, glContext);
+  CkSurface _makeSoftwareCanvasSurface(html.CanvasElement htmlCanvas) {
+    print('WARNING: failed to initialize WebGL. Falling back to CPU-only rendering.');
```

Comment:
Use `console.warn`. Also only send this error once. As written this will print this warning every time we resize the browser or add a platform view.

Reply:
Done.